src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -791,10 +791,6 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
       set_status("trimming bilog shards");
       yield call(new BucketTrimShardCollectCR(dpp, store, *pbucket_info, totrim.layout.in_index,
 					      min_markers));
-      // ENODATA just means there were no keys to trim
-      if (retcode == -ENODATA) {
-	retcode = 0;
-      }
```

Comment:
should this part clear ENOENT errors now?

Reply:
looks right, we would need the observer to have the bucket; otherwise, retrying on an empty shard perhaps would be useless.